### PR TITLE
improved clang-format

### PR DESCRIPTION
**Pull request by berniev**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [x ]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---
I have made a few changes to clang-format and am very pleased with the result.

I'm struggling to find any material issues. If there bits of code you want to keep manually formatted it is trivial to turn formatting off and on.

I don't think the current version on FC is being actively used, so it is a low risk option to merge this and encourage some feedback to get the ball rolling.

An agreed format is an easy productivity boost.

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112">pipeline status</a> for feature branch [PR_8112](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112). Pipeline [735669472](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/735669472) was triggered at [c492449](https://github.com/FreeCAD/FreeCAD/pull/8112/commits/c49244911b6883c2ec6cc92c5be9f43dfbbdd22d). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: .clang-format [hunk 1]
```diff
@@ -39,11 +39,13 @@ ContinuationIndentWidth: 4
 IndentCaseLabels: true
 IndentPPDirectives: None
 IndentWidth: 4
+InsertBraces: true
```
Comment (WandererFan) [inline .clang-format:42]: InsertBraces is problematic for clang-format-12 & Qt Creator.  Works nicely with 15.  Do we have a minimum version requirement for clang-format?

**Comment by WandererFan** [comment]
Doesn't most of the code base more or less follow "qt style"?  If so, should we use something like "https://github.com/qt-creator/qt-creator/blob/master/.clang-format" as our base instead of llvm?

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112">pipeline status</a> for feature branch [PR_8112](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112). Pipeline [735728181](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/735728181) was triggered at [c492449](https://github.com/FreeCAD/FreeCAD/pull/8112/commits/c49244911b6883c2ec6cc92c5be9f43dfbbdd22d). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berniev** [comment]
Yes, Insert Braces was introduced in v15. Which version people use doesn't really matter. What is critical is CI being the final fixer. Conceivably one can reformat to taste locally, confident that what actually appears in the project is to an agreed standard.

I tried a number of pre-rolled versions and they all fell short. This one was _based_ on llvm, tweaked. I found the base versions notable for what they do _not_ cover.

Running this version over old FC code produces many changes. For newer code it is more of a gentle tidy up. Styling in FC is very inconsistent, representing 20+ years of trends and individual preferences.

I'd be interested in your feedback after trying it.

**Comment by WandererFan** [comment]
Error in text formatting: /usr/bin/clang-format-15: /home/wile/Source/FreeCAD/.clang-format:42:15: error: invalid boolean
InsertBraces: Always
              ^~~~~~
Error reading /home/wile/Source/FreeCAD/.clang-format: Invalid argument

Should be: 
InsertBraces: true

**Comment by WandererFan** [comment]
If you are accepting artistic complaints, I really don't like this one:
  BeforeElse: true

I keep thinking I've found an else without an if.

**Comment by berniev** [comment]
Thanks for trying this out!
Re InsertBraces: I've checked here and checked the commit, and InsertBraces is set to True. Crossed wires somewhere?
Re: BeforeElse (assume you refer to BraceWrapping): I'm open to anything that gets us closer to concensus! Nothing wrong with your preference!

**Comment by WandererFan** [comment]
My bad.  I was swapping clang-formats in and out to do a PR.  

Is there any reason not to merge this?

**Comment by chennes** [comment]
Good by me -- Visual Studio uses the file to do its autoformatting, and this will take care of some issues I have with it.

**Comment by berniev** [comment]
On reflection, I am inclined to agree re turning off brace wrapping before else.
@chennes do you have a position either way?

**Comment by chennes** [comment]
Nope, I have no particular opinion either way. I've worked on projects that picked one, and projects that picked the other: as long as we pick I'm happy.

**Comment by berniev** [comment]
Force pushed amendment.

Now if you people could talk amongst yourselves and get this moving towards a standard ...

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112">pipeline status</a> for feature branch [PR_8112](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8112). Pipeline [738712013](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/738712013) was triggered at [2cd6fb1](https://github.com/FreeCAD/FreeCAD/pull/8112/commits/2cd6fb1709f678a0612ba761af2b7bb6fe80f6d7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: .clang-format [hunk 2]
```diff
@@ -4,28 +4,28 @@ AccessModifierOffset: -4
...
-AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: Always
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: Empty
 AllowShortCaseLabelsOnASingleLine: false
-AllowShortFunctionsOnASingleLine: All
+AllowShortFunctionsOnASingleLine: None
 AllowShortIfStatementsOnASingleLine: Never
-AllowShortLambdasOnASingleLine: All
+AllowShortLambdasOnASingleLine: Empty
```
Comment (donovaly) [inline .clang-format:12]: Why this change?

**Review by donovaly — APPROVED** [review]
Looks sensible for me too
